regular_expressions_basics/regular_expressions_basics.md

```diff
@@ -0,0 +1,338 @@
+* The space after the closing parenthesis can be added as-is:  ` `
+* The letters `claim` can be added just as they appear: `claim`
+* `:` is not a metacharacter for regex, so you can add it as is, with `:`, or, if you weren't sure if `:` was a metacharacter, you can also "escape" it, giving `\:`.
+
+This means that either `25\% \(20\) claim\:` or `25% \(20\) claim:` are regular expressions that match the given text.  On the other hand, `25% (20) claim:` and `25\% (20) claim\:` do not match the given text, because they fail to escape the parentheses.
+
+</div>
+***
+
+### Tokens: Special Groups
+
+Often, we want to indicate that a group of characters are all equally valid in a location.  For example, a pattern might include any digit at all in a particular location, or any of the letters "A", "Q", or "Y" at another location.  This is very typical, and this is where regular "find" or "find and replace" tools in software fail, while regular expressions prove useful.
+
+There are several ways to indicate groups of possible characters, and we'll go over the most common ones below.
+
+Metasequences
+------  
+
+Digits and some other groups are so common they are predefined with special **metasequences**.  
+
+* Any single digit (0-9) is represented `\d`.  
+* Any whitespace character like space or tab? That's `\s`.
+* Any character at all (a wildcard)? `.`
+
+You can combine these metasquences with what we've already learned.  For example, if the pattern we're trying to define includes a single digit, the letter Z, some other character that could be anything, and the letter J, we could put that in a regular expression as `\dZ.J`.
+
+Custom Character Sets
+------
+
+Other groups of characters can be defined by square brackets.  You can type each member of the character set between square brackets.
+
+* One of the letters `Y` or `N` could be written as `[YN]` (or `[NY]`)
+* Either the number 0 or the number 1? `[01]` (or `[10]`)
+
+As we've done before, you can combine these.  For example, if the pattern we're describing begins with a single digit, then a hyphen, then either a "Y" or "N", we would write that in regex as `\d\-[YN]`.
+
+Ranges
+-----
+
+You can also use square brackets to define a range of [Unicode characters](https://en.wikipedia.org/wiki/List_of_Unicode_characters), by giving the first character in the set, a hyphen, and the last character in the set.  For example, the lowercase letters appear in alphabetical order in Unicode, so we could indicate "the letter `c`, the letter `h`, and all the letters in between" by writing `[c-h]`.
+
+You can even put two or more ranges in square brackets, right next to one another.
+
+* Any digit 0-7 can appear, but not other digits? `[0-7]`
+* Any digit 0-3 or 7-9? `[0-37-9]` represents the two ranges.  Don't put a space between the ranges, because the space will be interpreted as "or, a space might appear here, too"!
```

Comment:
Maybe add a note here about remembering that this is defining a single character, so there is no confusion that you might mean the number "37" because that uses two characters.

Reply:
I agree, yes!

Reply:
I addressed this in a "care" box in the restructured page (into which I inserted a new example):

Remember that a token represents a **single character**, so it might be helpful to read something like `[025-8]` aloud one character at a time: "zero or two or five through eight", instead of saying to yourself "twenty five through eight," which is confusing!